pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -367,11 +366,33 @@ public SegmentPartitionConfig getSegmentPartitionConfig() {
   }
 
   public long getMinTime() {
-    return _minTime;
+    Long minTime = extractTimeValue(_minValueMap.get(_timeColumnName));
+    if (minTime != null) {
+      return minTime;
+    }
+    return Long.MAX_VALUE;
   }
 
   public long getMaxTime() {
```

Comment:
Seems this is used only in `HLRealtimeSegmentDataManager`?

Reply:
Yes.